src/jsi/jsi_init.hpp

```diff
@@ -18,6 +18,14 @@
 
 #pragma once
 
+#include <functional>
+
+namespace realm {
+namespace js {
+std::function<void()> flush_ui_queue;
```

Comment:
This needed to be moved to the header file so that it is visible to `jsi_function.hpp`, I guess this is what I missed and created some convoluted workaround for in the original!